virtual-host-gatherer/lib/gatherer/modules/VMware.py

```diff
@@ -117,10 +117,19 @@ def __process_node(self, node, output):
                 host_name = host.summary.config.name.split()[0]
                 mhz = (float(host.hardware.cpuInfo.hz) / float(1000 * 1000))
                 ram = (int(host.hardware.memorySize / (1024 * 1024)))
+
+                self.log.debug("Host identification for {0} -> UUID: {1} Vendor: {2} Serial Number: {3}".format(
+                    str(host),
+                    host.hardware.systemInfo.uuid,
+                    host.hardware.systemInfo.vendor,
+                    host.hardware.systemInfo.serialNumber
+                ))
+
                 output[host_name] = {
                     'type': 'vmware',
                     'name': host_name,
-                    'hostIdentifier': str(host),
+                    'hostIdentifier': host.hardware.systemInfo.uuid,
```

Comment:
Hmmm, I have bad feelings about this. If this is really the UUID of the hardware I fear that we might get duplicate values.
I saw already values like 0000-0000-0000-0000 in this field.
But this is a long time ago. Maybe big server for datacenters do not have this problem.

Reply:
Well, from the logs I gathered from the customer I never got such values. They were all good. I included to logging for this very specific reason though, so we can see and use the other values in case. The customer unfortunately hasn't yet done a major upgrade, so we still don't know if it actually fixes the issue.